src/data/polynomial.lean

```diff
@@ -2391,6 +2405,24 @@ calc derivative (f * g) = f.sum (λn a, g.sum (λm b, C ((a * b) * (n + m : ℕ)
 lemma derivative_eval (p : polynomial R) (x : R) : p.derivative.eval x = p.sum (λ n a, (a * n)*x^(n-1)) :=
 by simp [derivative, eval_sum, eval_pow]
 
+lemma is_coprime_of_is_root_of_eval_derivative_ne_zero {R : Type*} [field R]
+  (f : polynomial R) (r : R) (hf : f.is_root r) (hf' : f.derivative.eval r ≠ 0) :
```

Comment:
```suggestion
/-- If `f' r ≠ 0`, then `f / (X - r)` is coprime with `X - r`. Note that we do not assume `f r = 0`,
because `f / (X - r) = (f - f r) / (X - r)`. -/
lemma is_coprime_of_is_root_of_eval_derivative_ne_zero {R : Type*} [field R]
  (f : polynomial R) (r : R) (hf' : f.derivative.eval r ≠ 0) :
```

Reply:
We do assume `f r = 0` (otherwise it's false). It's the `(hf : f.is_root r)` assumption.

Reply:
Whoops, you're totally right! I thought I had a counterexample, but actually we don't use `hf` at all. I've fixed this, and added the docstring.

Reply:
Linter told me that you don't use `hf`. I explained it in a comment.